ui/desktop/electron-app/src/helpers/app-updater.js

```diff
@@ -19,6 +21,29 @@ if (debug && process.env.APP_UPDATER_CURRENT_VERSION) {
   currentVersion = process.env.APP_UPDATER_CURRENT_VERSION;
 }
 
+// Returns the real CPU architecture of the machine.
+const returnArchitectureToUpdate = () => {
+  const nodeArchitecture = process.arch;
+
+  if (nodeArchitecture === 'arm64') {
+    return nodeArchitecture;
+  }
+
+  try {
+    const runsOnRosetta = parseInt(
+      execSync('sysctl -in sysctl.proc_translated'),
+    );
+
+    if (runsOnRosetta) {
+      return 'arm64';
+    } else {
+      return nodeArchitecture; // amd64
```

Comment:
Can we move this to after the try catch so it always gets returned whether we error out or not? If it's arm64 it'll hit the normal return statement